internal/api/environments_test.go

```diff
@@ -32,6 +33,140 @@ func (s *EnvironmentControllerTestSuite) SetupTest() {
 	s.router = NewRouter(nil, s.manager)
 }
 
+func (s *EnvironmentControllerTestSuite) TestList() {
+	call := s.manager.On("List", mock.AnythingOfType("bool"))
+	call.Run(func(args mock.Arguments) {
+		call.ReturnArguments = mock.Arguments{[]runner.ExecutionEnvironment{}, nil}
+	})
+	path, err := s.router.Get(listRouteName).URL()
+	s.Require().NoError(err)
+
+	s.Run("with no Environments", func() {
+		recorder := httptest.NewRecorder()
+		request, err := http.NewRequest(http.MethodGet, path.String(), nil)
+		s.Require().NoError(err)
+		s.router.ServeHTTP(recorder, request)
+		s.Equal(http.StatusOK, recorder.Code)
+
+		var environmentsResponse ExecutionEnvironmentsResponse
+		err = json.NewDecoder(recorder.Result().Body).Decode(&environmentsResponse)
+		s.Require().NoError(err)
+		_ = recorder.Result().Body.Close()
+
+		s.Empty(environmentsResponse.ExecutionEnvironments)
+	})
+	s.manager.Calls = []mock.Call{}
+
+	s.Run("with fetch", func() {
+		recorder := httptest.NewRecorder()
+		query := path.Query()
+		query.Set("fetch", "true")
+		path.RawQuery = query.Encode()
+		request, err := http.NewRequest(http.MethodGet, path.String(), nil)
+		s.Require().NoError(err)
+
+		s.router.ServeHTTP(recorder, request)
+		s.Equal(http.StatusOK, recorder.Code)
+		s.manager.AssertCalled(s.T(), "List", true)
+	})
+	s.manager.Calls = []mock.Call{}
+
+	s.Run("with bad fetch", func() {
+		recorder := httptest.NewRecorder()
+		query := path.Query()
+		query.Set("fetch", "YouDecide")
+		path.RawQuery = query.Encode()
+		request, err := http.NewRequest(http.MethodGet, path.String(), nil)
+		s.Require().NoError(err)
+
+		s.router.ServeHTTP(recorder, request)
+		s.Equal(http.StatusBadRequest, recorder.Code)
+		s.manager.AssertNotCalled(s.T(), "List")
+	})
+}
+
+func (s *EnvironmentControllerTestSuite) TestGet() {
```

Comment:
Similar to above: Is the happy path (known environment, just without the `fetch` parameter) not interesting or already covered (e.g., in the `TestNomadEnvironmentManager_Get`)? A simple "Yes" is okay 🙂